### DISCUSSION TITLE
48khz on main output device vs 44.1khz on earphones

### DISCUSSION THREAD

**Discussion by davideberra** [google/oboe#1966] [Q&A] [answered]
Hello!
I have some of the users of my app complaining they listen a slowed down audio when they connect bluetooth earphones.

The scenario is the following:
I initialized the AudioStream like this:

```
    AudioStreamBuilder builder;
    oboe::Result result = builder.setSharingMode(oboe::SharingMode::Exclusive)
            ->setSampleRateConversionQuality(oboe::SampleRateConversionQuality::Medium)
            ->setChannelConversionAllowed(true)
            ->setFormatConversionAllowed(true)
            ->setPerformanceMode(oboe::PerformanceMode::LowLatency)
            ->setFormat(oboe::AudioFormat::I16)
            ->setChannelCount(kChannelCount)
            ->setDataCallback(mDataCallback)
            ->setErrorCallback(mErrorCallback)
            ->setAllowedCapturePolicy(oboe::AllowedCapturePolicy::All)
            ->openStream(mStream);
```

Frames per burst and frequency are taken from the calls of:

```
AudioManager::getProperty(AudioManager.PROPERTY_OUTPUT_FRAMES_PER_BUFFER)
AudioManager::getProperty(AudioManager.PROPERTY_OUTPUT_SAMPLE_RATE)
```

... and they are set during the onCreate phase of the app and are set like this:

```
	void oboe_set_sample_rate(int32_t sample_rate) {
		oboe::DefaultStreamValues::SampleRate = (int32_t) sample_rate;
	}

	void oboe_set_frames_per_burst(int32_t frames_per_burst) {
		oboe::DefaultStreamValues::FramesPerBurst = (int32_t) frames_per_burst;
	}
```

My app is also instructed to output pcm audio at the specified sample rate.

The problem arise when sample_rate value returned by the system is 48000hz (so oboe::DefaultStreamValues::SampleRate = 48000) BUT, when the earphones are connected, it seems they consuming audio at 44100hz resulting in a slowed down audio.

How should i manage this scenario? It's not clear to me

**Comment by robertwu1** [comment]
[PROPERTY_OUTPUT_SAMPLE_RATE](https://developer.android.com/reference/android/media/AudioManager#PROPERTY_OUTPUT_SAMPLE_RATE) is for a low latency stream on the device.

To get the actual sample rate of a stream, please call getSampleRate() on a stream once it's opened. In your case, your bluetooth stream should return 44100 and your app should handle this according.

**Comment by philburk** [comment, marked as answer]
The [PROPERTY_OUTPUT_SAMPLE_RATE](https://developer.android.com/reference/android/media/AudioManager#PROPERTY_OUTPUT_SAMPLE_RATE)
is not reliable. The actual default sample rate may be different on different peripheral or different stream settings.

Also oboe::DefaultStreamValues::SampleRate only affects the default rate for OpenSL ES, not AAudio.

If you need a particular sample rate to match the rate of your content then you need to specify it directly:

    builder.setSampleRate(contentSampleRate);
